parsl/tests/test_htex/test_manager_selector_by_block.py

```diff
@@ -0,0 +1,54 @@
+import time
+
+import pytest
+
+import parsl
+from parsl.app.app import bash_app, python_app
+from parsl.channels import LocalChannel
+from parsl.config import Config
+from parsl.configs.local_threads import config
+from parsl.executors import HighThroughputExecutor
+from parsl.executors.high_throughput.manager_selector import (
+    BlockIdManagerSelector,
+    ManagerSelector,
+)
+from parsl.launchers import WrappedLauncher
+from parsl.providers import LocalProvider
+from parsl.usage_tracking.levels import LEVEL_1
+
+
+def local_config():
+    return Config(
+        executors=[
+            HighThroughputExecutor(
+                label="htex_local",
+                max_workers_per_node=1,
+                manager_selector=BlockIdManagerSelector(),
+                provider=LocalProvider(
+                    channel=LocalChannel(),
+                    init_blocks=2,
+                    max_blocks=2,
+                    min_blocks=2,
+                ),
+            )
+        ],
+        usage_tracking=LEVEL_1,
+    )
+
+
+@parsl.python_app
+def get_worker_pid():
+    import os
+    return os.environ.get('PARSL_WORKER_BLOCK_ID')
+
+
+@pytest.mark.local
+def test_block_id_selection(try_assert):
+    blockids = []
+    try_assert(lambda: (time.sleep(1) or get_worker_pid().result() == "1"), timeout_ms=20000)
```

Comment:
whats the sleep(1) for? in test cases that's usually something that should be waited for explicitly, or is something broken that the author hopes goes away after a while.

Reply:
I think maybe you should be checking that *both* blocks are available when this test runs - eg `lambda: len(htex.connected_managers()) == 2` - so that there is the opportunity for buggy behaviour to place blocks onto block 0, and then the opportunity for the test to detect that.